docs/pages/setup/reference/cli.mdx

```diff
@@ -743,6 +745,46 @@ add_headers:
       "Authorization": "Bearer tokentokentoken"
 ```
 
+#### Aliases
+
+You can define custom `tsh` command aliases using the `aliases` section. It also allows for changing of the default flag values for existing commands.
+
+New command `tsh l`:
+
+```yaml
+aliases:
+    "l": "tsh login --auth=okta --username=alice"
+```
+
+Make `tsh status` use JSON as a default format:
+
+```yaml
+aliases:
+    "status": "tsh status --format=json"
+```
+
+Define custom command using with `bash`. The `$0` and `$1` variables will be substituted by command arguments.
+
+```yaml
+aliases:
+    "connect": "bash -c 'tsh login $0 && tsh ssh $1'"
+```
+
+The alias can use an arbitrary number of arguments. If a argument variable `$N` is referenced, `tsh` will check that at least `N+1` arguments were given to the alias invocation. All arguments that were given but not referenced in the alias definition will be appended at the end.
```

Comment:
Did we do sufficient testing to ensure we don't panic with an index out of bounds error when the wrong number of arguments are provided?

Reply:
Yep, there are unit tests for this: https://github.com/gravitational/teleport/blob/24fcadc375d8359e80790b3ebeaa36bd8dd2822f/tool/tsh/aliases_test.go#L49

But do let me know if you see something missing, happy to add more tests.